### PR TITLE
[O2-2424] DPL Analysis: Rework slicing algorithm for merged AODs

### PR DESCRIPTION
**Pull request by aalkin**
Solves O2-2424

### CODE REVIEW

File: Framework/Core/include/Framework/Kernels.h [hunk 1]
```diff
@@ -57,37 +58,52 @@ auto sliceByColumn(char const* key,
...
-    while (nextValue - value > 1) {
-      makeSlice(0);
-      ++value;
+  auto vprev = v;
+  auto vnext = v;
+  for (auto i = 0; i < size - 1; ++i) {
+    vprev = v;
+    v = values.Value(i);
+    count = counts.Value(i);
+    if (v >= 0) {
```
Comment (ktf) [inline Framework/Core/include/Framework/Kernels.h:79]: Frankly, I would rewrite things with less nesting:

```
for (...) {
if (v < 0) {
      vnext = values.Value(i + 1);
       while (vnext - vprev > 1) {
         makeSlice(offset, 0);
         ++vprev;
       }
       makeSlice(offset, count);
       offset += count;
       continue;
}
if (vprev < 0 || (v - vprev) == 1) {
         makeSlice(offset, count);
         offset += count;
         continue;
 }

while (v - vprev > 1) {
           makeSlice(offset, 0);
           ++vprev;
}
makeSlice(offset, count);
offset += count;
}
```

**Comment by sawenzel** [comment]
Thanks @aalkin. This solves the problem indeed.

**Review by sawenzel — DISMISSED** [review]

**Comment by aalkin** [comment]
@ktf @sawenzel I've found even better solution - we do not, in fact, need to keep unassigned groups available for normal grouping operations. The slicing function will optionally fill and return those in a separate vector which may be needed later for different tasks. This also allows us to return to previous version of grouping that does not have any runtime checks.

**Comment by sawenzel** [comment]
MacOS compilation checked locally.

**Pull request merged**